src/plots/cartesian/set_convert.js

```diff
@@ -575,13 +575,21 @@ module.exports = function setConvert(ax, fullLayout) {
 
     // should skip if not category nor multicategory
     ax.clearCalc = function() {
+        var init = function() {
```

Comment:
For this function name, it may be better to use something like `doClear` or `emptyCategories` since it is not initialised by default on top of the function.

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/3565/commits/598a67bce659e737ecb4dfe11a645135b60f31a5